### PR TITLE
<TagList/> - add size prop

### PR DESCRIPTION
**Pull request by gendeld**
### 🔦 Summary

Adds size prop to TagList to control Tag and ActionButton sizes.

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5743/

File: src/TagList/TagList.js [hunk 1]
```diff
@@ -46,6 +62,11 @@ TagListAction.propTypes = {
   dataHook: PropTypes.string,
   children: PropTypes.node,
   onClick: PropTypes.func,
+  size: PropTypes.oneOf(['tiny', 'small', 'medium']),
```
Comment (EgoziE) [inline src/TagList/TagList.js:65]: Remove size prop from here, it's not a public API

File: src/TagList/index.d.ts [hunk 2]
```diff
@@ -2,16 +2,17 @@ import * as React from 'react';
 import { TagProps } from '../Tag';
 import { ButtonProps } from '../Button';
 
+export type TagListSize = 'small' | 'medium' | 'large';
+
 export type TagListProps = {
   dataHook?: string;
   tags: Array<TagProps>;
```
Comment (EgoziE) [inline src/TagList/index.d.ts:9]: Not all `<Tag/>` props are public API in this case, please pick only those we expose.

File: src/TagList/TagList.js (hunk 1 above)
Comment (EgoziE) [reply src/TagList/TagList.js:65]: Also add `id` and `theme` and maybe more, check wha't relevant from `<Tag/>` component in this or another PR

File: src/TagList/index.d.ts [hunk 3]
```diff
@@ -2,16 +2,17 @@ import * as React from 'react';
...
+  size?: TagListSize;
```
Comment (EgoziE) [inline src/TagList/index.d.ts:10]: And also add it to the type test `src/TagList/test/TagList.tsx` as well

File: src/TagList/TagList.js [hunk 4]
```diff
@@ -57,11 +78,18 @@ TagList.propTypes = {
   /** List of tags props to be rendered */
   tags: PropTypes.arrayOf(PropTypes.object),
 
+  /** The size of each individual `<Tag />` */
+  size: PropTypes.oneOf(['small', 'medium', 'large']),
```
Comment (EgoziE) [inline src/TagList/TagList.js:82]: Please add a visual test for this prop

File: src/TagList/index.d.ts (hunk 2 above)
Comment (gendeld) [reply src/TagList/index.d.ts:9]: The code as written accepts any TagProp and works properly, shouldn't the types reflect that?

File: src/TagList/TagList.js (hunk 1 above)
Comment (gendeld) [reply src/TagList/TagList.js:65]: Can you explain what we add to propTypes for private components?

File: src/TagList/index.d.ts [hunk 5]
```diff
@@ -2,16 +2,17 @@ import * as React from 'react';
 import { TagProps } from '../Tag';
 import { ButtonProps } from '../Button';
 
+export type TagListSize = 'small' | 'medium' | 'large';
+
 export type TagListProps = {
   dataHook?: string;
-  tags: Array<TagProps>;
+  tags: Array<Omit<TagProps, 'size'>>;
```
Comment (splintor) [inline src/TagList/index.d.ts:9]: @EgoziE This use of `Omit<>` which [was added in TypeScript 3.5](https://www.infoq.com/news/2019/06/typescript-3-5-release/) broke [our build](http://tc.dev.wixpress.com/viewLog.html?buildId=118814619&buildTypeId=mono_repo_wixos_ping_settings_PingSettingsClient_O&tab=buildLog&_focus=169#_state=467), as we were still using `"typescript": "~3.0.1"`.

You might want to add a pre-req TypeScript version in https://github.com/wix/wix-style-react/blob/master/docs/FAQ/TYPES.MD

**Pull request merged**